python/shotgun_fields/shotgun_field_delegate.py

```diff
@@ -43,8 +53,13 @@ def _create_widget(self, parent):
         :returns:       QWidget that will be used to paint grid cells in the view.
         :rtype:         :class:`~PySide.QtGui.QWidget`
         """
-        self._display_widget.setParent(parent)
-        return self._display_widget
+        widget = self._display_class(
+            entity_type=self._entity_type,
+            field_name=self._field_name,
+            bg_task_manager=self._bg_task_manager,
+        )
+        widget.setParent(parent)
```

Comment:
Yep, I need to go back in and clean up the delegate class. I haven't really touched it since early on.